### PR TITLE
[FIX] point_of_sale: order total round globally

### PR DESCRIPTION
**Pull request by kebeclibre**
Before this commit, the amount total of an order in round_globally
was simply the sum of all taxes amounts, then rounded
This is plain wrong, as the accepted semantics of round globally is that:
- each tax for each line is computed without rounding
- then each taxes is summed across all order lines
- then that amount is rounded
- then all those rounded tax amount are summed together

After this commit, this semantics is implemented

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/point_of_sale/static/src/js/models.js [hunk 1]
```diff
@@ -2488,9 +2488,36 @@ exports.Order = Backbone.Model.extend({
...
+            // 1. For each tax, sum their amount across all order lines
+            // 2. Round that result
+            // 3. Sum all those rounded amounts
+            var groupTaxes = {};
+            this.orderlines.each(function (line) {
+                var taxDetails = line.get_tax_details();
+                var taxIds = Object.keys(taxDetails);
+                for (var t = 0; t<taxIds.length; t++) {
+                    var taxId = taxIds[t];
+                    if (!(taxId in groupTaxes)) {
```
Comment (pimodoo) [inline addons/point_of_sale/static/src/js/models.js]: maybe:
```suggestion
                    if (taxId in groupTaxes) {
                        groupTaxes[taxId] += taxDetails[taxId];
                    } else {
                        groupTaxes[taxId] = taxDetails[taxId];
                    }
```

File: addons/point_of_sale/static/src/js/models.js [hunk 2]
```diff
@@ -2488,9 +2488,36 @@ exports.Order = Backbone.Model.extend({
...
+            var sum = 0;
+            var taxIds = Object.keys(groupTaxes);
+            for (var j = 0; j<taxIds.length; j++) {
+                var taxAmount = groupTaxes[taxIds[j]];
+                sum += round_pr(taxAmount, this.pos.currency.rounding);
+            }
+            return sum;
+        } else {
+            return round_pr(this.orderlines.reduce((function(sum, orderLine) {
+                return sum + orderLine.get_tax();
```
Comment (pimodoo) [inline addons/point_of_sale/static/src/js/models.js]: shouldn't we round for each tax on a same line?

**Comment by pimodoo** [comment]
@kebeclibre LGTM

**Comment by kebeclibre** [comment]
robodoo r+

**Pull request closed without merging**